File: LeetCode_Marathons/LCM4/Medium/Increasing_Triplet_Subsequence/a.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        value1: int = 2147483647
        value2: int = 2147483647
        for num in nums:
            if num <= value1: value1 = num
            elif num <= value2: value2 = num
            else: return True
        return False

File: LeetCode_Marathons/LCM4/Medium/Increasing_Triplet_Subsequence/a.java
class Solution {
    public boolean increasingTriplet(int[] nums) {
        int value1 = 2147483647, value2 = 2147483647;
        for (int num : nums) {
            if (num <= value1) value1 = num;
            else if (num <= value2) value2 = num;
            else return true;
        }
        return false;
    }
}